crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -31,13 +35,13 @@ impl ProtocolFee {
                 if self.fee_policy_skip_market_orders {
                     vec![]
                 } else {
-                    vec![self.policy]
+                    self.policy.to_domain(quote).into_iter().collect()
```

Comment:
Currently we wouldn't notice if an order that has a fee policy that needs a quote doesn't get it.
Or is this not considered an error?
If you return a `Result` instead of an `Option` building the `vec` of fees while still being able to provide error messages would probably be easiest.

Reply:
Should we just log it and skip the problematic order or the entire function should return the error? https://github.com/cowprotocol/services/blob/80e0acbc45038fb3131c711dd8fff70b461f416b/crates/autopilot/src/solvable_orders.rs#L243

Reply:
My gut feeling for the time being is to only log and keep the order without fees. Once we are confident that the system works most of the time we can turn it into errors.